src/inmanta/graphql/schema.py

```diff
@@ -12,53 +12,24 @@
 Contact: code@inmanta.com
 """
 
+import dataclasses
 import typing
 
 import inmanta.data.sqlalchemy as models
 import strawberry
 from inmanta.data import get_session, get_session_factory
+from inmanta.server.services.compilerservice import CompilerService
 from sqlalchemy import Select, asc, desc, select
 from strawberry import relay
 from strawberry.schema.config import StrawberryConfig
 from strawberry.types import Info
 from strawberry.types.info import ContextType
 from strawberry_sqlalchemy_mapper import StrawberrySQLAlchemyLoader, StrawberrySQLAlchemyMapper
 
-mapper: StrawberrySQLAlchemyMapper[typing.Any] = StrawberrySQLAlchemyMapper()
-SCHEMA: strawberry.Schema | None = None
 
-
-def get_expert_mode(root: "Environment") -> bool:
-    assert hasattr(root, "settings")
-    return bool(root.settings.get("enable_lsm_expert_mode", False))
-
-
-@mapper.type(models.Environment)
-class Environment:
-    # Add every relation/attribute that we don't want to expose in our GraphQL endpoint to `__exclude__`
-    __exclude__ = [
-        "project_",
-        "agentprocess",
-        "code",
-        "compile",
-        "configurationmodel",
-        "discoveredresource",
-        "environmentmetricsgauge",
-        "environmentmetricstimer",
-        "notification",
-        "parameter",
-        "resource_persistent_state",
-        "unknownparameter",
-        "agent",
-    ]
-    is_expert_mode: bool = strawberry.field(resolver=get_expert_mode)
-
-
-def get_schema() -> strawberry.Schema:
-    global SCHEMA
-    if SCHEMA is None:
-        SCHEMA = initialize_schema()
-    return SCHEMA
+@dataclasses.dataclass
+class GraphQLMetadata:
+    compiler_service: CompilerService
```

Comment:
* I don't understand the name of this class. Metadata means data about data. But this class holds the compiler service, which is not data at all. I have the feeling this represents the context in which graphQL is executing. Would it be better to call this `GraphQLContext`?
* It could also use a docstring to say what is actually represents.